jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -0,0 +1,121 @@
+
+from traitlets import List, Unicode, Bool, Int
+
+from astropy.table import QTable
+from astropy.coordinates import SkyCoord
+from astroquery.sdss import SDSS
+
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import PluginTemplateMixin, ViewerSelectMixin
+
+__all__ = ['Catalogs']
+
+
+@tray_registry('imviz-catalogs', label="Catalog Search")
+class Catalogs(PluginTemplateMixin, ViewerSelectMixin):
+    template_file = __file__, "catalogs.vue"
+    catalog_items = List([]).tag(sync=True)
+    catalog_selected = Unicode("").tag(sync=True)
+    results_available = Bool(False).tag(sync=True)
+    number_of_results = Int(0).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.catalog_items = ["SDSS"]
+
+        if self.catalog_selected not in self.catalog_items:
+            self.catalog_selected = self.catalog_items[0]
+
+    def search(self):
+        # note to self: should I call clear before searching?
+
+        # gets the current viewer
+        viewer = self.viewer.selected_obj
+
+        # nothing happens in the case there is no image in the viewer
+        if viewer.state.reference_data is None:
+            self.results_available = False
+            self.number_of_results = 0
+            return
+        # obtains the center point of the current image and converts the point into sky coordinates
+        x_center = (viewer.state.x_min + viewer.state.x_max) * 0.5
+        y_center = (viewer.state.y_min + viewer.state.y_max) * 0.5
+        skycoord_center = viewer.state.reference_data.coords.pixel_to_world(x_center, y_center)
+
+        # obtains the viewer's zoom limits based on the visible layer
+        ny, nx = viewer.state.reference_data.shape
+        zoom_x_min = max(0, viewer.state.x_min)
+        zoom_x_max = min(nx, viewer.state.x_max)
+        zoom_y_min = max(0, viewer.state.y_min)
+        zoom_y_max = min(ny, viewer.state.y_max)
+        zoom_coordinate = viewer.state.reference_data.coords.pixel_to_world(
+            [zoom_x_min, zoom_x_min, zoom_x_max, zoom_x_max],
+            [zoom_y_min, zoom_y_max, zoom_y_max, zoom_y_min])
+
+        # radius for querying the region is based on:
+        # the distance between the longest zoom limits and the center point
+        zoom_radius = max(skycoord_center.separation(zoom_coordinate))
+
+        # conducts search based on SDSS
+        if self.catalog_selected == "SDSS":
+            # queries the region (based on the provided center point and radius)
+            # finds all the sources in that region
+            query_region_result = SDSS.query_region(skycoord_center, radius=zoom_radius,
+                                                    data_release=17)
+            self.results_available = True
+
+        else:
+            self.results_available = False
+            self.number_of_results = 0
+            raise ValueError(f"{self.catalog_selected} not a supported catalog")
```

Comment:
There is no point in resetting anything if you throw exception here.
```suggestion
            raise NotImplementedError(f"{self.catalog_selected} not a supported catalog")
```

But if you don't want a hard crash, then you can also instead opt to emit an error via snackbar and then you have to reset the results.

Reply:
This could only ever happen from the API, so isn't an error (after the reset so that the UI updates) appropriate?  Although I can see the argument that if its from the API, the UI shouldn't update at all and just stay in its previous state.

Reply:
I agree that throwing exception, as done here, is appropriate. Still not sure about resetting but I guess doesn't hurt to leave those in if you want.

Reply:
p.s. I still prefer `NotImplementedError` over `ValueError`.